gabarit/template_nlp/nlp_project/package_name/monitoring/mlflow_logger.py

```diff
@@ -36,28 +37,48 @@
 class MLflowLogger:
     '''Abstracts how MlFlow works'''
 
-    def __init__(self, experiment_name: str, tracking_uri: str = '') -> None:
+    def __init__(self, experiment_name: str, tracking_uri: str = '', artifact_uri: str = '') -> None:
         '''Class initialization
         Args:
             experiment_name (str):  Name of the experiment to activate
         Kwargs:
             tracking_uri (str): URI of the tracking server
+            artifact_uri (str): URI where to store artifacts
         '''
         # Get logger
         self.logger = logging.getLogger(__name__)
 
         # Backup to local save if no uri (i.e. empty string)
-        if tracking_uri == '':
+        if not tracking_uri:
             tracking_uri = pathlib.Path(os.path.join(utils.get_data_path(), 'experiments', 'mlruns')).as_uri()
+        
+        if not artifact_uri:
+            artifact_uri = pathlib.Path(os.path.join(utils.get_data_path(), 'experiments', 'mlruns_artifacts')).as_uri()
+
         # Set tracking URI & experiment name
         self.tracking_uri = tracking_uri
-        # No need to set_tracking_uri, this is done through the setter decorator
-        self.experiment_name = experiment_name
+        
+        # Get the experiment if it exists and check if there is a connection error by doing it
         try:
-            mlflow.set_experiment(experiment_name)
+            experiment = mlflow.get_experiment_by_name(experiment_name)
```

Comment:
Does that return an empty value if no experiment exists with this name ?

Reply:
Yes, it returns `None`.